sign.go

```diff
@@ -151,7 +151,11 @@ func (sd *SignedData) AddSignerChain(ee *x509.Certificate, pkey crypto.PrivateKe
 	h := hash.New()
 	h.Write(sd.data)
 	sd.messageDigest = h.Sum(nil)
-	encryptionOid, err := getOIDForEncryptionAlgorithm(pkey, sd.digestOid)
+	if sd.encryptionOid == nil {
+		// if the encryption algorithm wasn't set by SetEncryptionAlgorithm,
+		// infer it from the digest algorithm
+		sd.encryptionOid, err = getOIDForEncryptionAlgorithm(pkey, sd.digestOid)
```

Comment:
The original implementation used a local variable `encryptionOid` which did not otherwise persist the encryption algorithm OID beyond the current signing session.

Is it wise to set `sd.encryptionOid` (which is the equivalent to calling the public API func `SetEncryptionAlgorithm`). What if the `SignedData` is to be reused for a different encryption algorithm that is not explicitly specified? (Can it be reused?)